File: Product_APP_MVC/app/src/main/java/com/example/product_app_mvc/DB/DB_Creator.java
package com.example.product_app_mvc.DB;


import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

import com.example.product_app_mvc.model.POJO_class;

@Database(entities = {POJO_class.class}, version = 2, exportSchema = false)
@TypeConverters({Converters.class})  // Add this line
public abstract class DB_Creator extends RoomDatabase {
    private static DB_Creator instance = null;

    /*this abstract method that returns a ref to Fav_Product_DB_DAO
    * b/c Room will create an object of class that implements Fav_Product_DB_DAO interface
    * and returns it*/
    public abstract Fav_Product_DB_DAO getFav_Product_DB_DAO();

    public static synchronized DB_Creator getIlnstance(Context context) {
        if (instance == null) {
            instance = Room.databaseBuilder(context.getApplicationContext(), DB_Creator.class, "DB_Fav_Products")
                    .fallbackToDestructiveMigrationFrom(1, 2)
                    .build();
        }
        return instance;
    }

}

File: Android_Development/05_Android_Kotlin/06_Session/MyApplication/app/src/main/java/com/example/myapplication/database/DB_Creator.kt
package com.example.myapplication.database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.example.myapplication.model.Product

@Database(entities = [Product::class], version = 1, exportSchema = false)
abstract class DB_Creator : RoomDatabase() {

    abstract fun favProductDao(): Fav_Product_DB_DAO

    companion object {
        @Volatile
        private var INSTANCE: DB_Creator? = null

        fun getDatabase(context: Context): DB_Creator {
            return INSTANCE ?: synchronized(this) {
                val instance = Room.databaseBuilder(
                    context.applicationContext,
                    DB_Creator::class.java,
                    "products_database"
                ).build()
                INSTANCE = instance
                instance
            }
        }
    }
}
